util/website.py

```diff
@@ -0,0 +1,189 @@
+import os
+import paramiko
+import pysftp
+import sys
+
+from bs4 import BeautifulSoup
+from log import merge_logs
+
+def parse_for_deps(filename):
+    """
+    This takes an HTML file as input and output a list of existing depenencies.
+    Empty output means something was wrong.
+    Dependency is one of: js script loaded, css stylesheet loaded
+    """
+
+    with open(filename, 'r') as code:
+        soup = BeautifulSoup(code.read(), 'html.parser')
+
+    external_links   = filter(lambda x: x is not None, [ link.get('href') for link in soup.find_all('link') ])
+    external_scripts = filter(lambda x: x is not None, [ script.get('src') for script in soup.find_all('script') ])
+
+    # Verify with stat()
+    try:
+        all_resources = map(lambda x: os.stat(x), external_links + external_scripts)
+    except OSError as ex:
+        if ex.errno == 2:
+            print "Missing dependency", ex
+            return []
+        raise ex
+
+    # Try to read the file
+    try:
+        readable_resources = map(lambda x: os.open(x, os.O_RDONLY), external_links + external_scripts)
+        map(lambda x: os.close(x), readable_resources)
+    except OSError as ex:
+        print "Unreadable dependency", ex
+        return []
+
+    # It should be all good.
+    return external_links + external_scripts
+
+def get_ssh(auth_infos):
+    """
+    Connect to SSH server
+    """
+    cinfo = {'host':        auth_infos['ds_website_server_fqdn'],
+             'username':    auth_infos['ds_website_username'],
+             'private_key': auth_infos['ds_website_privkey'],
+             'port':        auth_infos['ds_website_server_port']}
+    return pysftp.Connection(**cinfo)
+
+def verify_ssh_dir(auth_infos):
+    """
+    This should ensure that SSH connection works and that the directory where
+    we want to push data is either empty or contains at least a .htaccess and
+    index.htm file
+    """
+    print "Checking connection: %s@%s:%s (port %d)" % (auth_infos['ds_website_username'], auth_infos['ds_website_server_fqdn'], auth_infos['ds_website_server_root'], auth_infos['ds_website_server_port'])
+
+    try:
+        with get_ssh(auth_infos) as sftp:
+            with sftp.cd(auth_infos['ds_website_server_root']):
+                files = sftp.listdir()
+                if len(files) == 0 or (('.htaccess' in files) and ('index.htm' in files)):
```

Comment:
The current website is indeed hosted on apache infra. That code is here to make sure that in this case, we push to something ready, since it looks like they (GoDaddy) don't have support for DirectoryIndex index.html by default (hence the .htaccess verification).